### DISCUSSION TITLE
Turbulent kinetic energy in HAFS's output

### DISCUSSION THREAD

**Discussion by zhup01** [hafs-community/HAFS#120] [Q&A]
I am trying to analyze the turbulent kinetic energy (TKE) simulated by HAFS. I understand that TKE is treated as a tracer in HAFS. I am running two versions of HAFS-A. One is a specific version that has TKE in HAFS output, but all values of TKE are NaN. The other version is the one that I checked out from git clone. But this one does not have TKE in the output. Can anyone show me how to correctly output TKE in HAFS's output? 

Thank you so much!

Ping

**Comment by ChunxiZhang-NOAA** [comment]
Hi Ping,

You can output TKE by adding the following line in the diag_table.tmp file under parm/forecast/regional if you run with workflow or diag_table in your run directory if you run testing cases in the run directory.
    "gfs_dyn",     "sgs_tke",        "tke",    "fv3_history",    "all",  .false.,  "none",  2

Thanks,
Chunxi

**Comment by BinLiu-NOAA** [comment]
@zhup01, In addition to the change mentioned by @ChunxiZhang-NOAA above, you will also need to change:
forecast.output_history=.true.
either in the command line option of ./run_hafs.py in your cronjob driver (or in the parm/hafs.conf file, etc.), so that the atmfhhh.nc and sfcfhhh.nc files will be written out in the forecast run directory (containing the TKE variable).

If you want to further output the TKE variable in the grib2 output files, you will need to further modify the post control files under [parm/post](https://github.com/hafs-community/HAFS/tree/develop/parm/post), for which @LinZhu-NOAA can provide you further instructions if needed.
